vitrina/structure/views.py

```diff
@@ -3424,34 +3435,212 @@ def form_valid(self, form):
+                        base=new_metadata_instance.object.base
+                        if isinstance(new_metadata_instance.object, Model)
+                        else None,
+                        status=new_metadata_instance.status if new_metadata_instance.status else None,
+                    )
+
+            already_created_fields = old_to_new_metadata_object_map
+            for old_related_instance, new_related_instance in list(old_to_new_metadata_object_map.items()):
+                try:
+                    already_created_fields = self.duplicate_foreign_key_relationships(
+                        new_related_instance, already_created_fields
+                    )
+                except ValidationError as e:
+                    transaction.set_rollback(True)
+                    form.add_error(None, e.message)
+                    return self.form_invalid(form)
+
+        if self.new_version:
+            return redirect(reverse("dataset-structure", args=[self.dataset.pk, self.new_version.pk]))
+        else:
+            return redirect(reverse("dataset-structure", args=[self.dataset.pk, self.metadata_version.pk]))
+
+    def create_related_model_duplicate(self, old_metadata_instance):
+        related_model = old_metadata_instance.content_type.model_class()
+        if related_model == Dataset:
+            return None
+        old_related_instance = related_model.objects.filter(id=old_metadata_instance.object_id).first()
+        new_related_instance = deepcopy(old_related_instance)
+        new_related_instance.pk = None
+        new_related_instance.metadata_version = self.new_version
+        new_related_instance.save()
+
+        return old_related_instance, new_related_instance
+
+    def create_metadata_duplicate(self, old_metadata_instance) -> tuple:
+        new_metadata_instance = deepcopy(old_metadata_instance)
+        new_metadata_instance.pk = None
+        # TODO: column draft can be deprecated after full versioning is completed.
+        new_metadata_instance.draft = False
+        if old_metadata_instance.status == Status.objects.filter(codename=StatusCode.DEVELOP).first():
+            new_metadata_instance.status = Status.objects.filter(codename=StatusCode.COMPLETED).first()
+
+        new_metadata_instance.metadata_version = self.new_version
+        new_metadata_instance.save()
+
+        return old_metadata_instance, new_metadata_instance
+
+    def duplicate_foreign_key_relationships(self, new_related_object, already_created_fields):
+        needed_foreign_key_relationships = [
+            Base,
+            Model,
+            Property,
+            EnumItem,
+            ParamItem,
+            DatasetDistribution,
+            Enum,
+            Param,
+            Prefix,
+        ]
+        for field in new_related_object._meta.get_fields():
+            if isinstance(field, ForeignKey):
+                # If the foreign key relationship is not in the needed list skip
```

Comment:
That's quite understandable from the code, no real value from this message, unless you think it improves readability, then ignore.